test/src/test/java/org/corfudb/runtime/clients/NettyCommTest.java

```diff
@@ -569,6 +573,141 @@ public void testTlsCipherRsaAndEcdsaKeyStoreEcdsaEcdsa() throws Exception {
+        serverCertManager.trustStoreManager.addCertificate(clientCertManager);
+        serverCertManager.trustStoreManager.save();
+        NettyClientRouter clientRouter = new NettyClientRouter(
+                NodeLocator.builder().host("localhost").port(port).build(),
+                buildRuntimeParams(clientCertManager.certManagementConfig)
+        );
+        clientRouter.getConnectionFuture().join();
+        assertThat(getBaseClient(clientRouter).pingSync()).isTrue();
+
+        // Reload ssl
+        clientRouter.reconnect();
+        TimeUnit.SECONDS.sleep(1);
+        assertThat(getBaseClient(clientRouter).pingSync()).isTrue();
+
+        clientRouter.close();
+
+        serverData.shutdownServer();
+        serverData.serverContext.close();
+    }
+
+    @Test
+    public void testReloadSslCertsWithCorruptedKeyStore() throws Exception {
+        int port = findRandomOpenPort();
+
+        Path certDir = Paths.get(PARAMETERS.TEST_TEMP_DIR);
+
+        // Start corfu server
+        CertificateManager serverCertManager = CertificateManager.buildSHA384withEcDsa(certDir);
+        NettyServerData serverData = new NettyServerData(buildServerContext(port, serverCertManager.certManagementConfig));
+        serverData.bootstrapServer();
+
+        // Happy path with correct cert
+        CertificateManager clientCertManager = CertificateManager.buildSHA384withEcDsa(certDir);
+        clientCertManager.trustStoreManager.addCertificate(serverCertManager);
+        clientCertManager.trustStoreManager.save();
+        serverCertManager.trustStoreManager.addCertificate(clientCertManager);
+        serverCertManager.trustStoreManager.save();
+        NettyClientRouter clientRouter = new NettyClientRouter(
+                NodeLocator.builder().host("localhost").port(port).build(),
+                buildRuntimeParams(clientCertManager.certManagementConfig)
+        );
+        clientRouter.getConnectionFuture().join();
+        assertThat(getBaseClient(clientRouter).pingSync()).isTrue();
+
+        // Corrupt client cert and reload ssl
+        Path keyStoreFilePath = clientCertManager.keyStoreConfig.getKeyStoreFile();
+        Path keyStoreFilePathCopy = keyStoreFilePath.resolveSibling(keyStoreFilePath.getFileName() + ".copy");
+        Files.move(keyStoreFilePath, keyStoreFilePathCopy, StandardCopyOption.ATOMIC_MOVE);
+
+        byte[] randomBytes = new byte[100];
+        Random random = new Random();
+        random.nextBytes(randomBytes);
+        Files.write(keyStoreFilePath, randomBytes, StandardOpenOption.CREATE_NEW, StandardOpenOption.WRITE);
+
+        clientRouter.reconnect();
+        TimeUnit.SECONDS.sleep(1);
+        assertThat(getBaseClient(clientRouter).pingSync()).isFalse();
+
+        // Restore the correct cert and reload ssl
+        Files.move(keyStoreFilePathCopy, keyStoreFilePath, StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.ATOMIC_MOVE);
+        clientRouter.reconnect();
```

Comment:
Same as above -- Wait for a timeout of `fileWatcherPollPeriod` instead?

Reply:
This tests against the reconnect() API where file watcher is not in picture. The test against file watcher is added in the latest patch.